### PR TITLE
initial user docs

### PR DESCRIPTION
**Pull request by lilyjazz**
## Why are these changes needed?

initial user docs

## Related issue number

NA

## Checks

- [x] I've included any doc changes needed for https://github.com/pingcap/LinguFlow/.
- [x] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [ ] I've made sure all auto checks have passed.

### CODE REVIEW

File: docs/user-guide/develop_builder_block.md [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+
+A Block represents a node within the Directed Acyclic Graph (DAG), each embodying a specific processing logic. Typically, a Block comprises several components:
+- **Block Name**: The name of the block.
+- **Block ID**: The unique identifier of the block.
+- **Copy Block ID**: A feature to copy the block's ID.
+- **Conditional Port**: Accepts True or False outputs from upstream `condition` type blocks, determining whether to execute this block based on the upstream result.
+- **Inport**: The information inputted into this block. Some blocks support multiple inports. Inports can handle `text`, `list`, `dict` data types.
+- **Add an Inport**: Adds an additional inport to the block. Manually added inports support connections of `any` data type.
+- **Outport**: The information outputted from this block. Outports typically support `text`, `list`, `dict`, `boolean` data types.
+- **Configuration**: Specific configuration details for the block, which vary from one block type to another.
```
Comment (shhdgit) [inline docs/user-guide/develop_builder_block.md:20]: According to https://github.com/pingcap/LinguFlow/pull/50/commits/d090e52b2f324777f6371c367346a932574999a3, we need to rename the Configuration -> Parameter.

File: docs/user-guide/develop_application_and_version.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
+---
+title: Self-Host Deployment
+sidebar_label: Self-Host
```
Comment (shhdgit) [inline docs/user-guide/develop_application_and_version.md:3]: Application & Version?

File: docs/user-guide/develop_builder_debugging.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
+---
+title: Block
+sidebar_label: Builder/Block
```
Comment (shhdgit) [inline docs/user-guide/develop_builder_debugging.md:3]: Debugging?

Comment (lilyjazz) [reply docs/user-guide/develop_builder_debugging.md]: done

File: docs/user-guide/develop_application_and_version.md (hunk 2 above)
Comment (lilyjazz) [reply docs/user-guide/develop_application_and_version.md]: done

File: docs/user-guide/develop_builder_block.md (hunk 1 above)
Comment (lilyjazz) [reply docs/user-guide/develop_builder_block.md]: done

**Review by shhdgit — APPROVED** [review]
lgtm.

**Pull request merged**